runtime/bcverify/rtverify.c

```diff
@@ -2456,6 +2456,14 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 		storeVerifyErrorData(verifyData, BCV_ERR_STACK_UNDERFLOW, (U_32)errorStackIndex, (UDATA)-1, (UDATA)-1, start);
 	}
 
+	/* Update the top index of the live stack with the current top index given it is
+	 * required to compute the number of elements on 'stack' in generating the detailed
+	 * error messages in pushLiveStackToVerificationTypeBuffer() at errormessageframeworkrtv.c,
+	 * especially when errorStackIndex points to the data type on 'locals' in terms of the
+	 * loading operations.
+	 */
+	SAVE_STACKTOP(liveStack, stackTop);
```

Comment:
Line 2455 uses the liveStack->stackElements value which is only updated here.  ie: `stackBase - liveStack->stackElements;` 

Should the SAVE_STACKTOP be done before that if statement?

Reply:
There is no difference to do so given the `if` statement relies on `stackTop` instead of `liveStack->stackTopIndex` which is updated with `stackTop` in `SAVE_STACKTOP`.
```
#define SAVE_STACKTOP(liveStack, stackTop) \
	liveStack->stackTopIndex = (IDATA) (stackTop - liveStack->stackElements)
```
But we can move it to the front of the `if` statement to avoid any confusion.

Reply:
Thanks, you're correct.  I think I had misread the `stackTopIndex` and `stackElements`.